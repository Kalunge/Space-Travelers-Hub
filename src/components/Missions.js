/* eslint-disable operator-linebreak */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  bookMission,
  getMissions,
  leaveMission,
} from '../redux/missions/missions';

const Missions = () => {
  const { missions } = useSelector((state) => state.missions);

  const dispatch = useDispatch();

  useEffect(() => {
    if (!missions.length) {
      dispatch(getMissions());
    }
  }, [dispatch, missions]);

  const joinMembershipHandler = (id) => {
    dispatch(bookMission(id));
  };

  const leaveMissionHandler = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <div className="mt-4 ">
      <table className="table table-bordered table-responsive table-striped">
        <thead>
          <tr>
            <th scope="col">Mission</th>
            <th scope="col">Description</th>
            <th scope="col">status</th>
            <th scope="col">{}</th>
          </tr>
        </thead>
        <tbody>
          {missions &&
            missions.map((item) => (
              <tr key={item.id}>
                <th scope="row">{item.name}</th>
                <td>{item.description}</td>
                {item.reserved ? (
                  <>
                    <td>
                      <span className="badge bg-primary mission-badge">
                        ACTIVE MEMBER
                      </span>
                    </td>
                    <td>
                      <button
                        type="button"
                        className="member-btn btn-danger"
                        data-testid="leave"
                        onClick={() => leaveMissionHandler(item.id)}
                      >
                        LEAVE MISSION
                      </button>
                    </td>
                  </>
                ) : (
                  <>
                    <td>
                      <span className="badge bg-dark mission-badge">
                        NOT A MEMBER
                      </span>
                    </td>
                    <td>
                      <button
                        onClick={() => joinMembershipHandler(item.id)}
                        type="button"
                        className="member-btn btn-light"
                        data-testid="join"
                      >
                        JOIN MISSION
                      </button>
                    </td>
                  </>
                )}
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
