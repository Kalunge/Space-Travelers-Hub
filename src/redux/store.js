import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rockets from './rockets/rockets';

const middleware = [thunk, logger];

const reducer = combineReducers({
  rockets,
});

const store = createStore(reducer, applyMiddleware(...middleware));

export default store;
