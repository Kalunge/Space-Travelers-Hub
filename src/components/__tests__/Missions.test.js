import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Missions from '../Missions';

describe('snapshot test for Missions Component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
