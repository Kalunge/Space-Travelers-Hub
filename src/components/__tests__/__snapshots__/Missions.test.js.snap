// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Missions Component renders correctly 1`] = `
<div
  className="mt-4 "
>
  <table
    className="table table-bordered table-responsive table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Mission
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          status
        </th>
        <th
          scope="col"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
